pkg/cluster/delete.go

```diff
@@ -458,6 +480,14 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		m.log.Printf("deleting cluster MSI certificate")
+		err = m.deleteClusterMsiCertificate(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
I'm assuming it likely doesn't matter, but are we sure that deleting the cluster MSI certificate before deleting the resources is safe to do, e.g. in a situation where the resource deletion fails incredibly early and a user continues to use the cluster as-is? 

Is there any explicit reason why we wouldn't want to delete the cluster MSI certificate after deleting the resources, or should we just move this deletion to after resource deletion just to be safe?

Reply:
That makes sense to me. I've made this change.